### PR TITLE
SOLR-17483: Cleanup zk-related logic in bin/solr.cmd

### PR DESCRIPTION
**Pull request by malliaridis**
https://issues.apache.org/jira/browse/SOLR-17483

# Description

With the migration to Java classes many parts of the scripts can be removed or simplified. This PR focuses on the ZK related parts.

# Solution

Removes completely all ZK related logic from `bin/solr.cmd` and fixes an issue with `AuthTool`.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [X] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/bin/solr.cmd [hunk 1]
```diff
@@ -263,7 +263,6 @@ IF "%1"=="assert" goto run_solrcli
 IF "%1"=="zk" goto run_solrcli
```
Comment (epugh) [inline solr/bin/solr.cmd:263]: Maybe for another jira, but the way `bin/solr` works is that all the commands go thorugh a check for if you are special..  For example, here is the auth check: https://github.com/apache/solr/blob/main/solr/bin/solr#L594 and then if you aren't start/stop/restart, you get the `run_solrcli` treatment: https://github.com/apache/solr/blob/main/solr/bin/solr#L629.   This removed a bunch of lines!

File: solr/bin/solr.cmd [hunk 2]
```diff
@@ -1207,244 +1206,7 @@ SHIFT
 SHIFT
 goto parse_config_args
 
-REM Clumsy to do the state machine thing for -d and -n, but that's required for back-compat
```
Comment (epugh) [inline solr/bin/solr.cmd:1210]: wow!  that's awesome!

**Review by epugh — APPROVED** [review]
LGTM.    An option to remove some more lines by reorganizing the checks....

**Comment by epugh** [comment]
i have seen the `keystore` tests occasionally fail in the bats test...

File: solr/bin/solr.cmd (hunk 1 above)
Comment (malliaridis) [reply solr/bin/solr.cmd:263]: I am looking into this right now. We have a lot of repetitive code in `bin/solr.cmd` that can be optimized if code blocks are rearranged. The way `bin/solr` is handling the special commands is probably the approach we gonna follow on the windows script as well.

**Comment by malliaridis** [comment]
> i have seen the keystore tests occasionally fail in the bats test...

Are the bats working / running on Windows? O.o I thought they do not work, at least not on my machine. :s

**Pull request merged**